### DISCUSSION TITLE
Selectively block email addresses

### DISCUSSION THREAD

**Discussion by ntnhon** [simple-login/app#406] [Feature Requests]
For not having to disable the whole alias

**Comment by gritcalculator** [comment]
Agreed, I would (also) appreciate the possibility to block a entire domain. Last night I received 700+ emails (spam), and due to catch-all, 700 aliases are created 😔

**Comment by anansii** [comment]
In general, the possibility to have a custom filter for every alias would be awesome.

In my case, I have 2 custom domains, one of them has my "standard mail address" which I started using 15+ years ago. That one alias gets all the spam and all the personal mails (so disabling it, is not really an option). And because I had to whitelist the SL domains all that spam goes directly to my mailbox. 

Usually, I would put all the spam senders that pass the spam filter in a special mail filter and reject future mails in the hope to dry this specific source out, but now all mails come from SL so that is not an option, because the reject would go to SL and not to the spammer.

Also, as a second use case, one would be able to setup all filtering and all custom domains with SL and be quite flexible in regards to the "backend".

I am willing to pay extra for such functionality (already use Premium).

**Reply by nguyenkims** [reply]
Do you think a custom filter set at account level can make sense? As we can have hundreds of aliases and we tend to use the same filter, having custom filters for each alias can be a hassle to manage.

**Reply by anansii** [reply]
I do, it would solve the issue where an alias does get spam but cannot be simply deactivated for some reason (like in my 15 year old main mail address).

I also think it might help you guys as well. If one alias gets a lot of spam the mail provider receiving these mails will think the spam is coming from SL (assumption). If these mails can be blocked before they leave SL, even better, I would say.

It doesn't even have to be per alias. If you use sieve it could be per account and each alias could be handled by a rule, I think. It wouldn't be that different from the filtering a regular mail provider has. But then again I don't know if your backend uses sieve at all.

In MailCow (and therefore Dovecot I think) you can configure pre-filters, post-filters and user-filters. So the global stuff goes into pre and post and each user/account can do his custom stuff inside the user-filter. Now, I'm not saying you should do it the same way but the idea does not sound so unusual to me.

Last but not least, I don't think a full sieve like filter is needed. A basic list of blocked senders would be a great start, then maybe in the next version add "block if subject contains [something]".

SimpleLogin is great for the "one service, one address" approach but handling legacy addresses while moving to that approach can be a pain.

**Reply by nguyenkims** [reply]
Yes I think for the first filter version, SimpleLogin will only support an exact match on whether the envelope sender or email from header. In the next versions, we can implement more advanced filters, e.g. based on regular expressions.

**Comment by gritcalculator** [comment]
I would appreciate an option to select multiple aliases based on a filter. Filter for example on aliases created the last 24 hours, select all and be able to disable those with 1 switch/click?

**Reply by nguyenkims** [reply]
Hey are you rather referring to this ticket? https://github.com/simple-login/app/discussions/407

**Reply by gritcalculator** [reply]
Hey, both to be honest 😄.. Aliases created (700+) were receiving e-mail
from one specific e-mail/domain. Now I had to switch off the catch-all.

**Comment by nguyenkims** [comment]
In the latest version, you can now block a too spammy sender without disabling the whole alias:

<img width="622" alt="Alias_Contact_Manager___SimpleLogin" src="https://user-images.githubusercontent.com/3167780/139222521-5dfbdcda-bb5c-4713-88a6-6748303d7987.png">
.

**Reply by TheLastProject** [reply]
You just made my day, I can finally stop rotating public contact emails on websites this often :)

**Comment by markliederbach** [comment]
Can there be an option to disable an entire sender's domain? I have an issue where I receive spam from different addresses within a set of domains. Short of having more complicated sieve-like filters (or trying to do this on client-side), it would be nice to disable an entire sender's domain.

**Reply by Gorson84** [reply]
I would also like that!

**Reply by IanNorris** [reply]
In the last few days I've started getting spam from two domains with lots of different subdomains (sa.com and ru.com) to aliases I can't rotate. It'd be nice if I could black hole future cases like these before they get to my client (or having to use a rule inside my provider's system).

This could work at the account, mailbox or alias level - I'm not picky, any would work for me!

**Reply by markliederbach** [reply]
Quick update, since I see that this is a decently backed idea, I've created a specific feature request for it. Please feel free to upvote it to give it some backing 🙏 

https://github.com/simple-login/app/discussions/1344
